app/src/main/cpp/skyline/services/nvdrv/core/nvmap.cpp

```diff
@@ -0,0 +1,144 @@
+            throw exception("Mapping nvmap handles without a CPU side address is unimplemented!");
+
+        size = util::AlignUp(size, PAGE_SIZE);
+        alignedSize = util::AlignUp(size, align);
+        address = pAddress;
+
+        // TODO: pin init
+
+        allocated = true;
+
+        return PosixResult::Success;
+    }
+
+    PosixResult NvMap::Handle::Duplicate(bool internalSession) {
+        // Unallocated handles cannot be duplicated as duplication requires memory accounting (in HOS)
+        if (!allocated) [[unlikely]]
+            return PosixResult::InvalidArgument;
+
+        std::scoped_lock lock(mutex);
+
+        // If we internally use FromId the duplication tracking of handles won't work accurately due to us not implementing
+        // per-process handle refs.
+        if (internalSession)
+            internalDupes++;
+        else
+            dupes++;
+
+        return PosixResult::Success;
+    }
+
+    NvMap::NvMap(const DeviceState &state) : state(state) {}
+
+    void NvMap::AddHandle(std::shared_ptr<Handle> handleDesc) {
+        std::scoped_lock lock(handlesLock);
+
+        handles.emplace(handleDesc->id, std::move(handleDesc));
+    }
+
+    bool NvMap::TryRemoveHandle(const std::shared_ptr<Handle> &handleDesc) {
+        // No dupes left, we can remove from handle map
+        if (handleDesc->dupes == 0 && handleDesc->internalDupes == 0) {
+            std::scoped_lock lock(handlesLock);
+
+            auto it{handles.find(handleDesc->id)};
+            if (it != handles.end())
+                handles.erase(it);
+
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    PosixResultValue<std::shared_ptr<NvMap::Handle>> NvMap::CreateHandle(u64 size) {
+        if (!size) [[unlikely]]
+            return PosixResult::InvalidArgument;
+
+        u32 id{nextHandleId.fetch_add(HandleIdIncrement, std::memory_order_relaxed)};
+        auto handleDesc{std::make_shared<Handle>(size, id)};
+        AddHandle(handleDesc);
+
+        return handleDesc;
+    }
+
+    std::shared_ptr<NvMap::Handle> NvMap::GetHandle(Handle::Id handle) {
+        std::scoped_lock lock(handlesLock);
+
+        try {
+            return handles.at(handle);
+        } catch (std::out_of_range &e) {
+            return nullptr;
+        }
+    }
+
+    std::optional<NvMap::FreeInfo> NvMap::FreeHandle(Handle::Id handle, bool internalSession) {
+        std::weak_ptr<Handle> hWeak{GetHandle(handle)};
+        FreeInfo freeInfo;
+
+        // We use a weak ptr here so we can tell when the handle has been freed and report that back to guest
+        if (auto handleDesc = hWeak.lock()) {
+            if (!handleDesc) [[unlikely]]
+                return std::nullopt;
+
+            std::scoped_lock lock(handleDesc->mutex);
+
+            if (internalSession) {
+                if (--handleDesc->internalDupes < 0)
+                    state.logger->Warn("Internal duplicate count inbalance detected!");
```

Comment:
It is

Reply:
I've read that it can be used as an alternative for imbalance but I cannot find any evidence of that in reputable dictionaries such as Merriam-Webster or the Cambridge dictionary.

https://dictionary.cambridge.org/us/spellcheck/english/?q=inbalance

Why not just use imbalance?